Dockerfile

```diff
@@ -17,37 +17,79 @@ ARG PG_MAJOR=13
 ## in relation to the total image size.
 ## By choosing a very basic base image, we do keep full control over every part
 ## of the build steps. This Dockerfile contains every piece of magic we want.
-FROM rust:1.54-slim-buster AS compiler
+
+## To allow us to use specific glibc 2.33+ features, we need to find a way
+## to run glibc 2.33. Running multiple glibc versions inside the same
+## container is something we'd like to avoid, we've seen multiple glibc
+## related bugs in our lifetime, adding multiple glibc versions in the mix
+## would make debugging harder.
+
+## Debian (and rust:debian) has served us well in the past, however even Debian's
+## latest release (bullseye, August 2021) cannot give us glibc 2.33.
+## Ubuntu however does give us glibc 2.33 - as Ubuntu is based upon Debian
+## the changes required are not that big for this Docker Image. Most of the
+## tools we use will be the same across the board, as most of our tools our
+## installed using external repositories.
+FROM ubuntu:21.04 AS compiler
 
 # We need full control over the running user, including the UID, therefore we
 # create the postgres user as the first thing on our list
 RUN adduser --home /home/postgres --uid 1000 --disabled-password --gecos "" postgres
 
-# CI/CD may benefit a lot from using a specific package mirror
-ARG DEBIAN_REPO_MIRROR=""
-RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend \
-    && if [ "${DEBIAN_REPO_MIRROR}" != "" ]; then \
-        sed -i "s{http://.*.debian.org{http://${DEBIAN_REPO_MIRROR}{g" /etc/apt/sources.list; \
-    fi
+RUN echo 'APT::Install-Recommends "0";\nAPT::Install-Suggests "0";' > /etc/apt/apt.conf.d/01norecommend
 
 # Install the highlest level dependencies
 RUN apt-get update \
-    && apt-get install -y curl ca-certificates locales gnupg1 lsb-release
+    && apt-get install -y curl ca-certificates gpg-agent wget gpg locales gnupg1 lsb-release
+
+RUN apt-get purge libappstream4
+
+RUN mkdir -p /build
+RUN chmod 777 /build
+WORKDIR /build/
 
+RUN wget -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor > /usr/share/keyrings/postgresql.keyring
```

Comment:
If a program has an option to write output, it's better to use that than redirection as redirection can leave empty files around if the program errors. In this case, you could use `gpg --dearmor -o /usr/share/keyrings/postgresql.keyring`

(not relevant here, but this also works better with sudo).